alot/commands/__init__.py

```diff
@@ -15,25 +15,44 @@ class Command(object):
 
     """base class for commands"""
     repeatable = False
+    mode = ""
+    cmdname = ""
 
     def __init__(self):
         self.undoable = False
         self.help = self.__doc__
 
-        class_ = type(self)
-        class_name = class_.__name__
-        cmdname, mode = reverse_lookup_command(class_)
-
-        # fetch and set pre and post command hooks
-        # they are set to None if not defined in the hooks file
-        get_hook = settings.get_hook
-        self.prehook = (get_hook('pre_%s_%s' % (mode, cmdname)) or
-                        get_hook('pre_global_%s' % cmdname))
-        self.posthook = (get_hook('post_%s_%s' % (mode, cmdname)) or
-                         get_hook('post_global_%s' % cmdname))
+        # set to False if no attemtp to fetch hooks from settings
+        self._prehook = False 
+        self._posthook = False
 
-        logging.debug("%s prehook: %s", class_name, str(self.prehook))
-        logging.debug("%s posthook: %s", class_name, str(self.posthook))
+    @property
+    def id(self):
+        """Mode and command name of instance as string."""
+        class_ = type(self)
+        return class_.mode + '.' + class_.cmdname
+
+    @property
+    def prehook(self):
+        """Command pre hook."""
+        if self._prehook == False:
```

Comment:
The reason for using `==` here was to distinguish between `None` and `False`. If set to `False`, `settings.get_hook()` has not been called. If `self._prehook` is `None`, `settings.get_hook()` has been called but no hook was found, so if it is `None`, that is actually what is supposed to be returned.

The property will not be read that many times, but the idea was to avoid the extra calls to `settings.get_hook()`. What do you think?